### PR TITLE
KT-16636 Remove parentheses after deleting the last unused constructor parameter

### PR DESCRIPTION
**Pull request by t-kameyama**
https://youtrack.jetbrains.com/issue/KT-16636

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/inspections/UnusedSymbolInspection.kt [hunk 1]
```diff
@@ -386,8 +386,15 @@ class SafeDeleteFix(declaration: KtDeclaration) : LocalQuickFix {
     override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
         val declaration = descriptor.psiElement.getStrictParentOfType<KtDeclaration>() ?: return
         if (!FileModificationService.getInstance().prepareFileForWrite(declaration.containingFile)) return
+        val primaryConstructorParameterList = (declaration.parent as? KtParameterList)?.takeIf { it.parent is KtPrimaryConstructor }
         ApplicationManager.getApplication().invokeLater(
-                { SafeDeleteHandler.invoke(project, arrayOf(declaration), false) },
+                {
+                    SafeDeleteHandler.invoke(project, arrayOf(declaration), null, false, null) {
+                        if (primaryConstructorParameterList != null && primaryConstructorParameterList.parameters.isEmpty()) {
```
Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/inspections/UnusedSymbolInspection.kt]: can be shortened to `if (primaryConstructorParameterList?.parameters?.isEmpty() == true)`

Comment (t-kameyama) [reply idea/src/org/jetbrains/kotlin/idea/inspections/UnusedSymbolInspection.kt]: Thank you, I will fix

Comment (t-kameyama) [reply idea/src/org/jetbrains/kotlin/idea/inspections/UnusedSymbolInspection.kt]: Fixed

**Comment by mglukhikh** [comment]
Accepted. The only point is we should not delete primary constructor for `expect class` in any case. But this is multi-platform stuff and it's easier for me to fix it myself. Thank you.

**Comment by mglukhikh** [comment]
Merged, thanks

**Pull request closed without merging**